buildSrc/src/main/kotlin/AarAndJarCompatible.kt

```diff
@@ -0,0 +1,20 @@
+package app.cash.paparazzi.gradle
+
+import org.gradle.api.artifacts.CacheableRule
+import org.gradle.api.attributes.AttributeCompatibilityRule
+import org.gradle.api.attributes.CompatibilityCheckDetails
+import org.gradle.api.attributes.LibraryElements
+
+@CacheableRule
+abstract class AarAndJarCompatible : AttributeCompatibilityRule<LibraryElements> {
```

Comment:
I believe compatibility rules can be avoided, if you create custom configurations set up with the appropriate variant attribute matching.  This would avoid messing with the existing configurations and is cleaner.

Reply:
Hmm, my gradle knowledge is lacking here with how variant attribute matching works with a custom configuration.

Reply:
I was able to get this to work. One difference in my current implementation is that child dependencies don't get resolved and transformed like they do with the compatibility rules.

I still think this method makes more logical sense from a configuration standpoint

Reply:
☝🏽 scratch that. I had set the `isTransitive` property of the configuration to be false. Setting it to true matches the desired implementation of the compatibility rules attempt.